addons/project/models/project.py

```diff
@@ -1850,7 +1850,9 @@ def default_get(self, default_fields):
             if project.analytic_account_id:
                 vals['analytic_account_id'] = project.analytic_account_id.id
         else:
-            vals['user_ids'] = [Command.link(self.env.user.id)]
+            user_ids = self._fields['user_ids'].convert_to_cache(vals.get('user_ids', []), self)
+            if self.env.user.id not in user_ids:
+                vals['user_ids'] = [Command.set(list(user_ids) + [self.env.user.id])]
```

Comment:
I wonder if you cannot do that instead. 🤔 
```diff
diff --git a/addons/project/models/project.py b/addons/project/models/project.py
index f6885a27d46..fc5049d8483 100644
--- a/addons/project/models/project.py
+++ b/addons/project/models/project.py
@@ -1857,9 +1857,8 @@ class Task(models.Model):
             if project.analytic_account_id:
                 vals['analytic_account_id'] = project.analytic_account_id.id
         else:
-            user_ids = self._fields['user_ids'].convert_to_cache(vals.get('user_ids', []), self)
-            if self.env.user.id not in user_ids:
-                vals['user_ids'] = [Command.set(list(user_ids) + [self.env.user.id])]
+            user_ids = vals.get('user_ids', [])
+            user_ids.append(Command.link(self.env.user.id))
 
         return vals
```
🙂

Reply:
@xavierbol , Right, Done!!. Thanks

Reply:
Wonderful thanks :)